api-analysis.md

```diff
@@ -0,0 +1,137 @@
+
+##### Actions
+
+1. No need to migrate ECF `application_id` as this information is already present in the attribute: `ecf_id` in the `applications` table in NPQ.
+2. `application_id` will be returned to LP via the [response of this endpoint](https://manage-training-for-early-career-teachers.education.gov.uk/api-reference/reference-v3.html#api-v3-npq-applications-id-accept-post-parameters):
+   1. For existing applications (previous to the split), [it will return ￼`ecf_id`￼ from the current ￼`applications`￼ table](https://github.com/DFE-Digital/npq-registration/blob/98cdef7b2e879588499a2cd4d3e72a087782c415/db/schema.rb#L33).
+   2. For new applications (after split), it will return a new `UUID` which will be also stored in the same `applications` table (`ecf_id`) field.
+3. Rename `ecf_id` to `application_lead_provider_id`, so we break the link between NPQ and ECF (ECF is no longer the owner of the identifiers).
+
+**Conclusions:**
+
+1. For existing `application_id` the endpoint in NPQ will continue managing / using the same `UUID` as ECF
+2. For new applications, NPQ will generate the a new `UUID` which will have the same format that previous `UUID` generated from ECF.
+3. The migrations of application identifiers will be transparent for Lead Providers
+
+##### #### API endpoints that receive an `participant_id`s
+
+> There is only one accepted application per course and participant
+
+```
+# Samples
+GET  /api/v3/participants/npq/{participant_id}/outcomes
+POST /api/v3/participants/npq/{participant_id}/outcomes
+PUT  /api/v3/participants/npq/{participant_id}/change-schedule
+PUT  /api/v3/participants/npq/{participant_id}/defer
+PUT  /api/v3/participants/npq/{participant_id}/resume
+PUT  /api/v3/participants/npq/{participant_id}/withdraw
+
+```
+
+The parameter `participant_id`:
+
+1. Is currently generated and managed in `ECF`
+2. `ECF` stores multiple participant identities per user, and has proven to be difficult to reason about. In NPQ we have decided to simplify them as NPQ requirements are simpler.
+3. To all effects, in NPQ,  the `participant_id` is the `user_id`.
+
+#### Actions
+
+1. No need to migrate `participant_id` as this information is already present in the attribute: `ecf_id` in the `users` table in NPQ (this attribute stores the user `ID` in `ECF`)
+2. The `participant_id` will be returned to LP via the endpoints
+   - For existing participants, we will return the `ecf_id` from the current `users` table.
+   - For new participants, we will return a new `UUID` which will be also stored in the `users` table (`ecf_id`) field.
+3. Rename `ecf_id` to `lead_provider_participant_id`, so we break the link between NPQ and ECF.
+4. Create a `Finder` in NPQ to locate an application using the course and the participant.
```

Comment:
discussed in person: add a bit more clarification on the resolver/lookup for the API and rules behind it